### PR TITLE
Add new letter upload step form and controller

### PR DESCRIPTION
**Pull request by zheileman**
This PR will introduce a new, simplified, letter upload to replace the current
documents checklist (dropzone uploader).

At the moment it is not wired up to the decision tree and thus is not visible to
the user, nor there are breaking changes with our current implementation, but
because the final PR might otherwise be too big, I decided to create at least 2
smaller PRs, being this one the first of those, to be easier to review.

The new form is very similar to the other steps we have where only one document
can be uploaded (i.e. `grounds for appeal`) with the difference it will maintain
the ability to check `having problems uploading` and go to the postal kick-out.
In summary, we remove the dropzone and replace it with a single file uploader,
and customise the view to talk about the specific letter needed to be uploaded.

https://www.pivotaltracker.com/story/show/148420071

### CODE REVIEW

File: app/controllers/concerns/document_attachable.rb [hunk 1]
```diff
@@ -25,6 +25,10 @@ def document_provided?
     tribunal_case&.documents(document_key)&.any? || document.present?
   end
 
+  def document_presence
```
Comment (digitalronin) [inline app/controllers/concerns/document_attachable.rb]: I'm not sure if this is the right name for this method. `document_presence` sounds like a method that would return a boolean, depending on whether a document was present or not. I would change this to `validate_document_presence`

Comment (zheileman) [inline app/controllers/concerns/document_attachable.rb]: Yes, I was unsure about this, but it seems to be the recommended naming, at least according to Rails documentation:

screen shot 2017-07-12 at 09 54 30

Or we can name it to something a bit more verbose, along the lines: `document_cannot_be_blank` if we follow this other example:

screen shot 2017-07-12 at 09 58 54

Comment (digitalronin) [inline app/controllers/concerns/document_attachable.rb]: Cool. But in that case, it feels to me like it belongs in a validator class.

Comment (zheileman) [inline app/controllers/concerns/document_attachable.rb]: IMO for this simple validation, shuffling around the 1 line of code to another class (validator) will introduce a level of indirection that will make the code more difficult to follow.
I will refactor this and see how it looks, but might feel a bit over engineered tbh.

**Comment by zheileman** [comment]
@digitalronin So, I introduced a Validator for the presence which, IMO, had a couple of negative side effects:

* The validator was not being autoloaded so I had to add the instruction in the `application.rb` (the other option, equally ugly, was to require it explicitly in the Form object). At least this solution works if we were to have more validators in the future.
* Had to make public 2 methods that previously were private, because the Validator needs access to them.

What do you think?

N.B: I've already ranted enough about **Danger** in the past but again, here we have one of their negatives things IMO: this Validator refactor should have been added as a separate commit to the branch, to be easier to see what were the actual changes, but instead, to **please** Danger, I had to squash both commits together, making it difficult to actually see how the code evolves from one commit to another. But, just my opinion.

File: app/views/steps/closure/support_documents/edit.html.erb [hunk 2]
```diff
@@ -23,7 +23,7 @@
       <div class="form-group util_mt-large">
         <%= f.label :having_problems_uploading, class: 'block-label selection-button-checkbox', 'data-target': 'unable_to_upload_panel' do %>
           <%= f.check_box :having_problems_uploading, 'aria-controls': 'unable_to_upload_panel' %>
-          <%= t('shared.file_upload.having_problems') %>
+          <%= t('.having_problems') %>
```
Comment (digitalronin) [inline app/views/steps/closure/support_documents/edit.html.erb]: Should this be `having_problems_[something]`?
